### PR TITLE
Ph pd 2514 multiome on terra

### PR DESCRIPTION
**Pull request by phendriksen100**
### Description

Updated Multiome.wdl and atac.wdl to include a `cloud_provider` parameter. This enhancement enables us to specify the preferred cloud service provider for workflow execution, determining the appropriate Docker images and storage file locations accordingly.

[PD-2514](https://broadworkbench.atlassian.net/browse/PD-2514)
----

### Checklist 
If you can answer "yes" to the following items, please add a checkmark next to the appropriate checklist item(s) **and** notify our WARP documentation team by tagging either @ekiernan or @kayleemathews in a comment on this PR.

- [x] Did you add inputs, outputs, or tasks to a workflow?
- [ ] Did you modify, delete or move: file paths, file names, input names, output names, or task names?
- [x] If you made a changelog update, did you update the pipeline version number?

[PD-2514]: https://broadworkbench.atlassian.net/browse/PD-2514?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pipelines/skylab/multiome/Multiome.wdl [hunk 1]
```diff
@@ -25,33 +26,56 @@ workflow Multiome {
...
     }
 
+    # Determine docker prefix based on cloud provider
+    String gcr_docker_prefix = "us.gcr.io/broad-gotc-prod/"
+    String acr_docker_prefix = "dsppipelinedev.azurecr.io/"
+    String docker_prefix = if cloud_provider == "gcp" then gcr_docker_prefix else acr_docker_prefix
+
+    # Define docker images
+    snap_atac_docker_image = "snapatac2:1.0.4-2.3.1-1700590229"
+
```
Comment (phendriksen100) [inline pipelines/skylab/multiome/Multiome.wdl:56]: Will need to update this when Nareh's warp tools PR goes threw

File: pipelines/skylab/multiome/Multiome.wdl [hunk 2]
```diff
@@ -25,33 +26,56 @@ workflow Multiome {
...
+    # Define all whitelist files
+    File gcp_gex_whitelist = "gs://gcp-public-data--broad-references/RNA/resources/arc-v1/737K-arc-v1_gex.txt"
+    File gcp_atac_whitelist = "gs://gcp-public-data--broad-references/RNA/resources/arc-v1/737K-arc-v1_atac.txt"
+    File azure_gex_whitelist = "https://datasetpublicbroadref.blob.core.windows.net/dataset/RNA/resources/arc-v1/737K-arc-v1_gex.txt"
+    File azure_atac_whitelist = "https://datasetpublicbroadref.blob.core.windows.net/dataset/RNA/resources/arc-v1/737K-arc-v1_atac.txt"
+
+    Determine which whitelist files to use based on cloud provider
```
Comment (nikellepetrillo) [inline pipelines/skylab/multiome/Multiome.wdl:63]:
```suggestion
    # Determine which whitelist files to use based on cloud provider
```

File: pipelines/skylab/multiome/atac.wdl [hunk 3]
```diff
@@ -91,21 +113,24 @@ workflow ATAC {
         output_base_name = input_id,
         nthreads = num_threads_bwa, 
         mem_size = mem_size_bwa,
-        cpu_platform = cpu_platform_bwa
+        cpu_platform = cpu_platform_bwa,
+        docker_path = docker_prefix + sam_tools_docker
```
Comment (nikellepetrillo) [inline pipelines/skylab/multiome/atac.wdl:117]:
```suggestion
        docker_path = docker_prefix + samtools_docker
```

File: pipelines/skylab/multiome/Multiome.wdl [hunk 4]
```diff
@@ -25,33 +26,56 @@ workflow Multiome {
...
 
     }
 
+    # Determine docker prefix based on cloud provider
+    String gcr_docker_prefix = "us.gcr.io/broad-gotc-prod/"
+    String acr_docker_prefix = "dsppipelinedev.azurecr.io/"
+    String docker_prefix = if cloud_provider == "gcp" then gcr_docker_prefix else acr_docker_prefix
+
+    # Define docker images
+    snap_atac_docker_image = "snapatac2:1.0.4-2.3.1-1700590229"
```
Comment (nikellepetrillo) [inline pipelines/skylab/multiome/Multiome.wdl:55]:
```suggestion
    String snap_atac_docker_image = "snapatac2:1.0.4-2.3.1-1700590229"
```

File: pipelines/skylab/multiome/Multiome.wdl (hunk 2 above)
Comment (phendriksen100) [reply pipelines/skylab/multiome/Multiome.wdl]: Oof, good catch

File: pipelines/skylab/multiome/Multiome.wdl (hunk 1 above)
Comment (sahakiann) [reply pipelines/skylab/multiome/Multiome.wdl:56]: Just merged the warp tools PR! The new tag is `1.0.5-2.3.2-1709230223` (see [here](https://console.cloud.google.com/gcr/images/broad-gotc-prod/us/snapatac2@sha256:56172a1e5c914c0839e55404ceeecc2876321b4f806e3475b6124836e4794e75/details?authuser=0&project=broad-gotc-prod)

File: pipelines/skylab/multiome/atac.wdl [hunk 5]
```diff
@@ -3,6 +3,7 @@ version 1.0
 import "../../../tasks/skylab/MergeSortBam.wdl" as Merge
 import "../../../tasks/skylab/FastqProcessing.wdl" as FastqProcessing
 import "../../../tasks/skylab/PairedTagUtils.wdl" as AddBB
+"../../../tasks/broad/Utilities.wdl" as utils
```
Comment (nikellepetrillo) [inline pipelines/skylab/multiome/atac.wdl:6]:
```suggestion
import "../../../tasks/broad/Utilities.wdl" as utils
```

**Comment by kayleemathews** [comment]
@phendriksen100 this will require minor doc updates

**Comment by nikellepetrillo** [comment]
closing this as it is now part of this pr: https://github.com/broadinstitute/warp/pull/1239

**Pull request closed without merging**